code/modules/reagents/reagent_containers/iv_bag.dm

```diff
@@ -0,0 +1,180 @@
+#define IV_DRAW 0
+#define IV_INJECT 1
+
+/obj/item/reagent_containers/iv_bag
+	name = "\improper IV Bag"
+	desc = "A bag with a fine needle attached at the end for injecting patients with fluids over a period of time."
+	icon = 'icons/goonstation/objects/iv.dmi'
+	lefthand_file = 'icons/goonstation/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/goonstation/mob/inhands/items_righthand.dmi'
+	icon_state = "ivbag"
+	volume = 200
+	possible_transfer_amounts = list(1,5,10)
+	amount_per_transfer_from_this = 1
+	container_type = OPENCONTAINER
+	var/label_text
+	var/mode = IV_INJECT
+	var/mob/living/carbon/human/injection_target
+
+/obj/item/reagent_containers/iv_bag/Destroy()
+	end_processing()
+	return ..()
+
+/obj/item/reagent_containers/iv_bag/on_reagent_change()
+	update_icon()
+
+/obj/item/reagent_containers/iv_bag/pickup(mob/user)
+	..()
+	update_icon()
+
+/obj/item/reagent_containers/iv_bag/dropped(mob/user)
+	..()
+	update_icon()
+
+/obj/item/reagent_containers/iv_bag/attack_self(mob/user)
+	..()
+	mode = !mode
+	update_icon()
+
+/obj/item/reagent_containers/iv_bag/attack_hand()
+	..()
+	update_icon()
+
+/obj/item/reagent_containers/iv_bag/proc/begin_processing(mob/target)
+	injection_target = target
+	processing_objects.Add(src)
+
+/obj/item/reagent_containers/iv_bag/proc/end_processing()
+	injection_target = null
+	processing_objects.Remove(src)
+
+/obj/item/reagent_containers/iv_bag/process()
+	if(!injection_target)
+		end_processing()
+		return
+
+	if(get_dist(get_turf(src), get_turf(injection_target)) > 1)
+		to_chat(injection_target, "<span class='userdanger'>The [src]'s' needle is ripped out of you!</span>")
+		injection_target.apply_damage(3, BRUTE, pick("r_arm", "l_arm"))
+		end_processing()
+		return
+
+	if(mode) 	// Injecting
+		if(reagents.total_volume)
+			var/fraction = min(amount_per_transfer_from_this/reagents.total_volume, 1) 	//The amount of reagents we'll transfer to the person
+			reagents.reaction(injection_target, INGEST, fraction) 						//React the amount we're transfering.
+			reagents.trans_to(injection_target, amount_per_transfer_from_this)
+			update_icon()
+	else		// Drawing
+		if(reagents.total_volume < reagents.maximum_volume)
+			injection_target.transfer_blood_to(src, amount_per_transfer_from_this)
+			for(var/datum/reagent/x in injection_target.reagents.reagent_list) // Pull small amounts of reagents from the person while drawing blood
+				injection_target.reagents.trans_to(src, amount_per_transfer_from_this/10)
+			update_icon()
+
+/obj/item/reagent_containers/iv_bag/attack(mob/living/M, mob/living/user, def_zone)
+	return
+
+/obj/item/reagent_containers/iv_bag/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!target.reagents)
+		return
+
+	var/mob/living/L
```

Comment:
since this is only ever used after the `isliving` check, it should go under there.